packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -88,9 +87,8 @@ exports.sourceNodes = async (
   let syncToken
   if (
     !pluginConfig.get(`forceFullSync`) &&
-    store.getState().status.plugins &&
-    store.getState().status.plugins[`gatsby-source-contentful`] &&
-    store.getState().status.plugins[`gatsby-source-contentful`][
+    store.getState()?.status?.plugins &&
+    store.getState()?.status?.plugins[`gatsby-source-contentful`][
```

Comment:
I think this isn't the same?

It was doing 
```
a.b().c.d && a.b().c.d[e] && a.b().c.d[e][f]
```
and you changed it to 
```
a.b().c && a.b().c.d && a.b().c.d[e][f]
```

```suggestion
    store.getState().status.plugins?.[`gatsby-source-contentful`]?.[
```

Reply:
it isn't but this one is more robust

Reply:
Can you elaborate? In particular, assumptions are made of things existing that were explicitly checked before. Before `store.getState()?.status?.plugins[`gatsby-source-contentful`]` was checked to exist before trying to access a property on it, afterwards it's apparently sufficient to confirm `plugins` exists? This may be true, but it's not obvious.

Also, you can merge the `store.getState()?.status?.plugins &&` of line 90 into into a `?.` on line 91, or drop the redundant occurrences of `?.` in the repeated part in line 91.

```suggestion
    store.getState()?.status?.plugins?.[`gatsby-source-contentful`][
```